### DISCUSSION TITLE
Bug Report

### DISCUSSION THREAD

**Discussion by Fred-Smashing** [FakeFishGames/Barotrauma#15241] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Circuit boxes causing several issues.

When using circuit boxes in a custom sub we experience several issues.

Before the unto the breach update, all aspects of the custom sub were working as intended.

No error messages or crashes, are produced when experiencing these issues.

Multiplayer and Singleplayer

- ~When un-docking in a campaign, using the docking hatch prefab with a curcuit box the sub gets stuck attatched to the docking hatch of the station.~ (This issue is a side effect of the ballast not responding)
- The ballasts are controlled by a circuit box that allows overidding the ballast to drain it. The ballasts also automatically drain when docked. When un-docking, the pumps do not work until we manually toggle the pumps using the lever, regardless of the lever state before un-docking.
    - Changing to a wired connection from a wifi component to get the docking state the issue persists.
    - Re-creating the same set of components outside of a circuit box and the issue is fixed.

Multiplayer Only
4. When a client uses the airlock with a circuit box controlling the logic the airlock gets stuck open for the host. Using the same airlock setup not in a circuit box the airlock behaves as expected.

### Reproduction steps

[The Leaky Robin T1-3.zip](https://github.com/user-attachments/files/17881873/The.Leaky.Robin.T1-3.zip)
](url)

1. Load the attatched sub
2. Use the airlocks/ballasts/docking in multiplayer
4. Experience one of the issues mentioned above.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by pavel4win** [comment]
I encountered the fact that the "or" component now working in different way.
Before the last update, if you run a wire through "or component" from the motion detector to the door, it would open when the player approached.
Now door clicks continuously, as if signals 0 and 1 were coming form "or" element in at maximum speed.
This effect can be eliminated by setting the "or" component's action time greater than 0. At least 0.01
Therefore, some circuits made in earlier versions of the game stopped working and require some fixes.

**Reply by Fred-Smashing** [reply]
Tried adjusting the Timeframe for the OR components in all my circuits. It does not fix any of the issues I'm experiencing.
I also tried re-creating the scenario you mentioned and couldn't re-produce that behaviour either, would be useful if you could send a screenshot of you're example so I can test that too?

**Reply by pavel4win** [reply]
Component "or" I gave as an example.
Try to set the timeframe greater than 0 (0.01 will be enough I think) in all components that have a timeframe ("or", "and", "greater", etc.)
Another non-obvious fact:
An empty output from the "memory" component is interpreted by other components input as a logical 1. 
I don't know how it was in previous versions of the game.
When I find time for this, I will try to give examples of situations in which the components began to behave differently than before. I will shoot a video or something like that.
Later I will look at the ballast drainage system on your boat.
I developed a circuit on 8 components, which works fine both in the box and without it.
It is tied to the state of the upper and lower docking hatch, the ballast is drained if one of them is open. Therefore, it does not require any lever. Later I will send a screenshot of the circuit.

**Reply by pavel4win** [reply]
The experiment proved the timeframe 0.01 is not enough. The timeframe must be 0.1 at last.
